utils/transaction-utils.js

```diff
@@ -397,107 +398,348 @@ const signAndPublishSlpTransaction = async (
       SLP.Address.detectAddressNetwork(fromAddr)
     );
-    let originalAmount: number = 0;
 
-    // Add all UTXOs to the transaction inputs.
-    for (let i = 0; i < utxos.length; i++) {
-      const utxo: utxo = utxos[i];
-      originalAmount = originalAmount + utxo.satoshis;
-      transactionBuilder.addInput(utxo.txid, utxo.vout);
-    }
+    if (tokenId && hasBCH) {
+      // The paper wallet has both BCH and SLP balances
+      // This case sweeps 1 SLP token and all of the BCH to the users wallet
+      // In the case the paper wallet has more than 1 SLP token, additional sweeps in the SLP only use case must be called
+
+      const scaledTokenSendAmount = new BigNumber(
+        balancesByKey[tokenId]
+      ).decimalPlaces(tokenDecimals);
+      const tokenSendAmount = scaledTokenSendAmount.times(10 ** tokenDecimals);
+      const sendOpReturn = slpjs.Slp.buildSendOpReturn({
+        tokenIdHex: tokenId,
+        outputQtyArray: [tokenSendAmount]
+      });
+      const tokenReceiverAddressArray = [addressSlp];
+
+      const slpUtxos = [...utxosByKey[tokenId]];
+      const bchUtxos = [...utxosByKey["BCH"]];
+
+      let inputUtxos = [...slpUtxos, ...bchUtxos];
+
+      let byteCount = SLPJS.calculateSendCost(
+        sendOpReturn.length,
+        inputUtxos.length,
+        tokenReceiverAddressArray.length + 1, // +1 to receive remaining BCH
+        fromAddr
+      );
 
-    if (originalAmount < 1) {
-      throw new Error(`Original amount is zero. No BCH to send.`);
-    }
+      let totalUtxoAmount = 0;
+      inputUtxos.forEach(utxo => {
+        transactionBuilder.addInput(utxo.txid, utxo.vout);
+        totalUtxoAmount += utxo.satoshis;
+      });
 
-    // get byte count to calculate fee. paying 1.1 sat/byte
-    const byteCount: number = SLP.BitcoinCash.getByteCount(
-      { P2PKH: utxos.length },
-      { P2PKH: 1 }
-    );
-    const fee: number = Math.ceil(1.1 * byteCount);
+      const satoshisRemaining = totalUtxoAmount - byteCount;
 
-    // amount to send to receiver. It's the original amount - 1 sat/byte for tx size
-    const sendAmount: number = originalAmount - fee;
+      // SLP data output
+      transactionBuilder.addOutput(sendOpReturn, 0);
 
-    // add output w/ address and amount to send
-    transactionBuilder.addOutput(
-      SLP.Address.toLegacyAddress(toAddr),
-      sendAmount
-    );
+      // Token destination output
+      transactionBuilder.addOutput(addressSlp, 546);
 
-    // Loop through each input and sign it with the private key.
-    let redeemScript;
-    for (let i: number = 0; i < utxos.length; i++) {
-      const utxo = utxos[i];
-      transactionBuilder.sign(
-        i,
-        keyPair,
-        redeemScript,
-        transactionBuilder.hashTypes.SIGHASH_ALL,
-        utxo.satoshis
+      // Return remaining bch balance output
+      // What is the purpose of the + 546 here again.  Without it the fee is way too high, just not sure why as fee already calculated
```

Comment:
If you are asking, I believe it's because the 546 for the output is already counted so we add it back now.

Reply:
Right, was just a thought I forgot to delete afterward.  Will remove it from source the next release